src/modals/edit-profile/profile-form/ProfileForm.js

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable react/jsx-handler-names */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { Form, Field } from 'react-final-form';
+import { ButtonPromiseState } from '../../../shared/components/button-promise-state';
+import nationalities from 'ms-nationalities';
+import { TextInput, Button, AutocompleteSelect, Radio, AvatarPicker, UserIcon } from '@nomios/web-uikit';
+import { notEmpty } from '../../../shared/form-validators';
+import LocationInput from './location-input';
+import styles from './ProfileForm.css';
+
+const PLACEHOLDER = {
+    name: 'Enter your name',
+    nationality: 'Enter your nationality',
+    location: 'Enter your location',
+};
+
+class ProfileForm extends Component {
+    state = {
+        image: null,
+        promise: undefined,
+    };
+
+    render() {
+        const { profileInfo } = this.props;
+        const { promise, image } = this.state;
+
+        return (
+            <Form
+                initialValues={ profileInfo }
+                onSubmit={ this.handleSubmit }>
+                { ({ handleSubmit, invalid, dirty, form }) => (
+                    <form className={ styles.form } autoComplete="off" onSubmit={ handleSubmit }>
+                        <div className={ styles.inputsContainer }>
+                            <div className={ styles.leftContent }>
+                                <div className={ classNames(styles.field, styles.name) }>
+                                    <label className={ styles.label }>Name</label>
+                                    <Field name="name" validate={ notEmpty }>
+                                        { ({ input }) => <TextInput { ...input } placeholder={ PLACEHOLDER.name } /> }
+                                    </Field>
+                                </div>
+                                <div className={ classNames(styles.field, styles.nationality) }>
+                                    <label className={ styles.label }>Nationality</label>
+                                    <Field name="nationality">
+                                        { ({ input }) => (
```

Comment:
This is just a preference, but given you're using `react-final-forms`, you can easily extract this into a separate render function to prevent a very large render function

Reply:
What are you suggesting? Extract this field (nationality) specifically? Or every field?